### ISSUE TITLE
Compatibility with Unitful Quantity types

### ISSUE THREAD

**Issue by mikeingold** [brainandforce/CliffordNumbers.jl#26]
Following up from #25, I would love to be able to use this package with `Unitful.Quantity` types. The following MWE demonstrates a usage that I would like to have work, but that currently doesn't.

```julia
using CliffordNumbers: KVector, VGA, ∧
using LinearAlgebra: norm
using Unitful

a = KVector{1, VGA(3)}(1.0u"m", 0.0u"m", 0.0u"m")
b = KVector{1, VGA(3)}(0.0u"m", 2.0u"m", 0.0u"m")
c = KVector{1, VGA(3)}(0.0u"m", 0.0u"m", 3.0u"m")

norm(a ∧ b ∧ c)
# -> 6.0u"m^3"
```

**Comment by brainandforce** [comment]
One thing I'll note here is the restriction of the scalars of `AbstractCliffordNumber` subtypes to `Union{Real,Complex}`. This is very much intentional: it would be a bad idea to use a `Quaternion` or an `AbstractCliffordNumber` as the scalar field when Clifford algebras are restricted to real or complex scalars. `Unitful.Quantity` does not subtype `Real` or `Complex` so it cannot be used as an element type for an `AbstractCliffordNumber`.

To get around this, you can just wrap a `KVector` or other `AbstractCliffordNumber` directly in a `Quantity`, since any subtype of `Number` is allowed to be wrapped:
```julia-repl
julia> using CliffordNumbers, Unitful

julia> KVector{1,VGA(3)}(1, 2, 3)u"m"
3-element KVector{1, VGA(3), Int64}:
1e₁ + 2e₂ + 3e₃ m

julia> typeof(ans)
Quantity{KVector{1, VGA(3), Int64, 3}, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}
```
(There is a printing issue here since `Base.show(::IO, ::Quantity)` calls `Base.show(::IO, ::AbstractCliffordNumber)`, which then calls `Base.summary(::IO, ::AbstractCliffordNumber)`, so the summary for the wrapped type is printed instead of the one for the type itself.)

**Comment by brainandforce** [comment]
I still need to include support for a few other operations, but your use case should be covered by this initial package extension.

**Comment by mikeingold** [comment]
Thanks! I’m looking forward to trying it out.

**Comment by brainandforce** [comment]
Final update: this has been released as of 0.1.9.
